src/Mvc/Mvc.Testing/src/WebApplicationFactory.cs

```diff
@@ -197,21 +302,54 @@ private void EnsureServer()
         {
             SetContentRoot(builder);
             _configuration(builder);
-            _server = CreateServer(builder);
+            if (_useKestrel)
+            {
+                _webHost = CreateKestrelServer(builder);
+
+                TryExtractHostAddress(() => GetServerAddressFeature(_webHost));
+            }
+            else
+            {
+                _server = CreateServer(builder);
+            }
+        }
+    }
+
+    private void TryExtractHostAddress(Func<IServerAddressesFeature?> serverAddressFeatureAccessor)
+    {
+        var serverAddressFeature = serverAddressFeatureAccessor();
```

Comment:
Nit:

```suggestion
    private void TryExtractHostAddress(IServerAddressesFeature? serverAddressFeature)
    {
```

Of course, the call sites will need to be updated too.

Reply:
Looking at the code now, I am wondering why I chose to go with the Func<> approach in the first place. Thanks for spotting this, @halter73, will fix.